src/bindings/interface/annotator.i

```diff
@@ -0,0 +1,217 @@
+%module(package="libcellml") annotator
+
+#define LIBCELLML_EXPORT
+
+%import "createconstructor.i"
+%import "enums.i"
+%import "logger.i"
+%import "types.i"
+
+%include <std_pair.i>
+%include <std_vector.i>
+
+%feature("docstring") libcellml::Annotator
+"Interacts with CellML objects using their id attribute.";
+
+%feature("docstring") libcellml::Annotator::setModel
+"Set the model for this Annototor to work with.";
+
+%feature("docstring") libcellml::Annotator::item
+"Return the item with the given id.";
+
+%feature("docstring") libcellml::Annotator::component
+"Return the component with the given id.";
+
+%feature("docstring") libcellml::Annotator::encapsulation
+"Return the model with the given encapsulation id.";
+
+%feature("docstring") libcellml::Annotator::model
+"Return the model assigned to this Annotator or the model with the given id.";
+
```

Comment:
Is this too ambiguous?  Should we have separate functions for these two things?

Reply:
'Tis the beauty of function overloading.

Reply:
Yeah .. but it sure ain't beautiful here :| The two functions are doing two totally different things!

Reply:
I am

Reply:
What happens if the current model in the annotator doesn't have the id specified?  Which function is it calling then?  I really don't like this.

Reply:
@hsorby and @kerimoyle: you guys need to come some kind of an agreement so that we can resolve this conversation. :)

Reply:
> What happens if the current model in the annotator doesn't have the id specified? Which function is it calling then? I really don't like this.

Isn't this just pointing to the overloaded C++ code, so as with the C++ code if you ask for a model with a given id, its only going to return a non-null pointer if the id matches that of the current model. Right? The only difference here is that there is a fall back option to provide the current model for the annotator...

I don't see a need to separate this into two separate python methods. (assuming I'm understanding this issue correctly)

Reply:
You have the same understanding that I have and I am happy with the way it works.

Reply:
I am happy with this too, so resolving.